torchdynamo/bytecode_transformation.py

```diff
@@ -121,9 +163,19 @@ def devirtualize_jumps(instructions):
                     break
 
             if inst.opcode in dis.hasjabs:
-                inst.arg = target.offset
+                if sys.version_info < (3, 10):
+                    inst.arg = target.offset
+                else:
+                    # arg is offset of the instruction line rather than the bytecode
+                    # for all jabs/jrel since python 3.10
+                    inst.arg = int(target.offset / 2)
             else:  # relative jump
-                inst.arg = target.offset - inst.offset - instruction_size(inst)
+                if sys.version_info < (3, 10):
+                    inst.arg = target.offset - inst.offset - instruction_size(inst)
+                else:
+                    inst.arg = int(
+                        (target.offset - inst.offset - instruction_size(inst)) / 2
+                    )
```

Comment:
Actually this took the most time of the overall work, I don't find any doc describing this change. Is this because it's Python internal changes which is not expected to expose users?

Reply:
Glad you figure this out.  Did you look in the Python 3.10 release notes?  I'd be surprised if it weren't mentioned there.

Reply:
I didn't find it in Python 3.10 release notes, that what surprised me. I'll open an issue at cython to ask why they make this change and if they document them anywhere.